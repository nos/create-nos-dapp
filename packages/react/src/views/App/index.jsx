import React from "react";
import injectSheet from "react-jss";
import PropTypes from "prop-types";

import Header from "./../../components/Header";
import NOSActions from "./../../components/NOSActions";

const styles = {
  "@import": "https://fonts.googleapis.com/css?family=Source+Sans+Pro",
  "@global html, body": {
    fontFamily: "Source Sans Pro",
    margin: 0,
    padding: 0,
    backgroundColor: "#ffffff"
  },
  App: {
    textAlign: "center"
  },
  intro: {
    fontSize: "large"
  },
  lineBreak: {
    width: "75%",
    borderTop: "1px solid #333333",
    margin: "32px auto"
  }
};

const App = ({ classes }) => (
  <div className={classes.App}>
    <Header title="A nOS dApp starter-kit!" />
    <p className={classes.intro}>
      To get started, edit <code>src/views/App/index.js</code> and save to reload.
    </p>
    <p className={classes.intro}>Or test out the following demo functions!</p>
    <hr className={classes.lineBreak} />
    <NOSActions />
  </div>
);

App.propTypes = {
  classes: PropTypes.object.isRequired
};

export default injectSheet(styles)(App);
