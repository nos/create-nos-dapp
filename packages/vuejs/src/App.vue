<template>
    <div id="app">
        <div v-if="nos.exists">
            <app-header :title="title" :nos="nos"></app-header>
            <router-view :nos="nos"/>
        </div>
        <div v-else>
            Launch me through nOS client, please!
        </div>
    </div>
</template>

<script>
import AppHeader from './components/app-header'
import nosApi from '@nosplatform/api-functions'

export default {
  name: 'app',
  components: { AppHeader },
  created () {
  },
  data: function () {
    return {
      nos: nosApi,
      title: 'nOS dapp'
    }
  }
}
</script>

<style lang="css">
</style>
