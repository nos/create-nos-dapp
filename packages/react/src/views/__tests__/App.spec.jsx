import React from "react";
import { render } from "enzyme";

import App from "./../App";

describe("Components", () => {
  describe("Header", () => {
    test("should render", () => {
      const wrapper = render(<App />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
