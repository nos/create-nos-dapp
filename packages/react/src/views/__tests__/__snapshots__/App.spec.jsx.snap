// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Header should render 1`] = `
<div
  class="App-App-0-1-1"
>
  <header
    class="Header-header-0-1-4"
  >
    <h1
      class="Header-title-0-1-5"
    >
      A nOS dApp starter-kit!
    </h1>
    <div
      class="SpinningLogo-logoContainer-0-1-6"
    >
      <img
        alt="nOS logo"
        class="SpinningLogo-nosLogo-0-1-7"
        src="test-file-stub"
      />
    </div>
  </header>
  <p
    class="App-intro-0-1-2"
  >
    To get started, edit 
    <code>
      src/views/App/index.js
    </code>
     and save to reload.
  </p>
  <p
    class="App-intro-0-1-2"
  >
    Or test out the following demo functions!
  </p>
  <hr
    class="App-lineBreak-0-1-3"
  />
  <button
    class="NOSActions-button-0-1-8"
  >
    Get Address
  </button>
  <button
    class="NOSActions-button-0-1-8"
  >
    Get NEO Balance
  </button>
  <button
    class="NOSActions-button-0-1-8"
  >
    Claim Gas
  </button>
  <button
    class="NOSActions-button-0-1-8"
  >
    TestInvoke (NeoAuth)
  </button>
  <button
    class="NOSActions-button-0-1-8"
  >
    GetStorage (NeoBlog)
  </button>
</div>
`;
